### PR TITLE
Add support for parameter mappers

### PR DESCRIPTION
**Pull request by daschult**
ms-rest-js changes necessary for https://github.com/Azure/autorest.typescript/issues/94.

### CODE REVIEW

File: lib/serializer.ts [hunk 1]
```diff
@@ -156,10 +158,10 @@ export class Serializer {
       } else if (typeName.match(/^Stream$/ig) !== null) {
         const objectType = typeof value;
         if (objectType !== "string" &&
-            objectType !== "function" &&
-            !(value instanceof ArrayBuffer) &&
-            !ArrayBuffer.isView(value) &&
-            !(typeof Blob === "function" && value instanceof Blob)) {
+          objectType !== "function" &&
```
Comment (RikkiGibson) [inline lib/serializer.ts]: I'd prefer if this either were reindented or a blank line were inserted above the `throw` statement in order to make it clear where the condition ends and the body begins.

File: lib/serializer.ts [hunk 2]
```diff
@@ -729,7 +731,7 @@ export interface Mapper {
   xmlIsWrapped?: boolean;
   readOnly?: boolean;
   isConstant?: boolean;
-  required: boolean;
+  required?: boolean;
```
Comment (RikkiGibson) [inline lib/serializer.ts]: poetic

File: test/shared/axiosHttpClientTests.ts [hunk 3]
```diff
@@ -3,9 +3,9 @@
 import * as assert from "assert";
 import * as should from "should";
 import { AxiosHttpClient } from "../../lib/axiosHttpClient";
-import { baseURL } from "../testUtils";
+import { isNode } from "../../lib/util/utils";
```
Comment (RikkiGibson) [inline test/shared/axiosHttpClientTests.ts]: sorting?

File: lib/serializer.ts (hunk 1 above)
Comment (daschult) [reply lib/serializer.ts]: I'll insert the blank line. I agree with you about the indentation, but the VS Code TypeScript formatter made it this way.

File: test/shared/axiosHttpClientTests.ts (hunk 3 above)
Comment (daschult) [reply test/shared/axiosHttpClientTests.ts]: According to the VS Code TypeScript extension's import sorting algorithm, this is the correct sorting order.

**Pull request merged**